### PR TITLE
[SYCL][NFC] Code cleanup revealed by self-build.

### PR DESCRIPTION
**Pull request by zahiraam**
Remove redundant false case in switch statement following:
https://llvm.org/docs/CodingStandards.html#don-t-use-default-labels-in-fully-covered-switches-over-enumerations

### CODE REVIEW

File: clang/lib/Sema/SemaSYCL.cpp [hunk 1]
```diff
@@ -3477,7 +3481,6 @@ static const char *paramKind2Str(KernelParamKind K) {
     CASE(std_layout);
     CASE(sampler);
     CASE(pointer);
-  default:
     return "<ERROR>";
```
Comment (premanandrao) [inline clang/lib/Sema/SemaSYCL.cpp:3484]: Shouldn't the return "<ERROR>" statement be moved outside the switch block?

**Review by kbobrovs — DISMISSED** [review]
BTW, I was wondering how to guard against forgetting to add a handling for a new enum value when it is added. Clang is consistent here and also issues an error in such case `error: enumeration value 'green' not handled in switch [-Werror,-Wswitch]` . 

post-link approved

File: clang/lib/Sema/SemaSYCL.cpp (hunk 1 above)
Comment (zahiraam) [reply clang/lib/Sema/SemaSYCL.cpp:3484]: Good catch! Updated it. Thanks.

File: clang/lib/Sema/SemaSYCL.cpp [hunk 2]
```diff
@@ -3477,9 +3481,8 @@ static const char *paramKind2Str(KernelParamKind K) {
     CASE(std_layout);
     CASE(sampler);
     CASE(pointer);
-  default:
-    return "<ERROR>";
   }
+    return "<ERROR>";
```
Comment (premanandrao) [inline clang/lib/Sema/SemaSYCL.cpp:3485]: Please fix the clang-format error about indentation here.    But otherwise, looks good.

**Comment by premanandrao** [comment]
That switch-case macro change in SemaSYCL.cpp has now disappeared.   Did you split that out?

**Comment by zahiraam** [comment]
> That switch-case macro change in SemaSYCL.cpp has now disappeared. Did you split that out?

I had a conflict resolution and I have done it on this page! May be I should fix that in my ws.

**Comment by zahiraam** [comment]
Can someone please review this? Thanks.

**Comment by elizabethandrews** [comment]
LGTM but code owners @AlexeySachkov @kbobrovs needs to approve.

**Review by premanandrao — APPROVED** [review]

**Comment by premanandrao** [comment]
> Can someone please review this? Thanks.

Sorry @zahiraam, I missed getting back to this after the intermittent test failure issue.

**Comment by zahiraam** [comment]
> > Can someone please review this? Thanks.
> 
> Sorry @zahiraam, I missed getting back to this after the intermittent test failure issue.

No worries. I have more reviews coming and wanted this one to be merged in before. Thanks.

**Review by AlexeySachkov — APPROVED** [review]

**Review by v-klochkov — COMMENTED** [review]
Minor comment: here and in future commits/PRs: please add '-s' option to 'git commit' command to add a signature the the commit message.

**Comment by bader** [comment]
> Minor comment: here and in future commits/PRs: please add '-s' option to 'git commit' command to add a signature the the commit message.

FYI: it's **not** required by https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md anymore.

**Pull request merged**